cprover_bindings/src/goto_program/expr.rs

```diff
@@ -1388,6 +1409,16 @@ impl Expr {
         expr!(StringConstant { s }, Type::c_char().array_of(s.len() + 1)).array_to_ptr()
     }
 }
+
+/// Wrap expression in a lambda
+impl Expr {
+    pub fn as_lambda_expression(&self, typ: Type, variables: &Vec<Expr>) -> Expr {
+        assert!(variables.iter().all(|x| x.is_symbol()), "Variables must be symbols");
+        let v = Expr::tuple_expr(Type::Empty, variables.to_vec());
```

Comment:
Is `Type::Empty` intentional? If so, can you add a comment to explain why?

Reply:
Removed `LambdaExpression` from `Expr`. Adding tuple at the `Irep` level now.